src/Microsoft.FeatureManagement.AspNetCore/TargetingHttpContextMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.FeatureManagement.FeatureFilters;
+using System;
+using System.Threading.Tasks;
+
+namespace EvaluationDataToApplicationInsights.Telemetry
+{
+    /// <summary>
+    /// Used to add targeting information to http context. This allows synronous code to access targeting information.
+    /// </summary>
+    public class TargetingHttpContextMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+
+        private const string TargetingIdKey = $"Microsoft.FeatureManagement.TargetingId";
+
+        /// <summary>
+        /// Creates an instance of the TargetingHttpContextMiddleware
+        /// </summary>
+        public TargetingHttpContextMiddleware(RequestDelegate next, ILoggerFactory loggerFactory) {
+            _next = next;
```

Comment:
check `next` for null

Reply:
Adjusted.

Didn't realize next would ever be null. Nearly all examples I see, the middleware completely assumes it wont be null, but looking into standard middlewares (like routing and authentication), I see they throw when it's null. Just surprised null handling isn't anywhere in the docs on writing middleware.

Reply:
Always check for null if null isn't accounted for. Regardless of anything else. In this case, the consumption of _next does not check for null so it should be validated in the constructor.